cmd/snap/cmd_services.go

```diff
@@ -184,8 +185,57 @@ func (s *svcLogs) Execute(args []string) error {
 	return nil
 }
 
+type userAndScopeMixin struct {
+	System bool   `long:"system"`
+	User   bool   `long:"user"`
+	Users  string `long:"users" optional:"yes" optional-value:"*"`
+}
+
+var userAndScopeDescs = mixinDescs{
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"system": i18n.G("The operation should only affect system services."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"user": i18n.G("The operation should only affect user services for the current user."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"users": i18n.G("The operation should affect all user services."),
+}
+
+func (um *userAndScopeMixin) serviceScope() (client.ScopeSelector, error) {
+	switch {
+	case um.User && um.System:
+		return nil, fmt.Errorf("--user and --system cannot be used in conjunction with each other")
+	case um.Users != "" && um.User:
+		return nil, fmt.Errorf("--user and --users cannot be used in conjunction with each other")
+	case um.Users != "" && um.System:
```

Comment:
Shouldn't  `um.Users == "*" && um.System` be allowed?

Reply:
yep, done!